skfp/model_selection/splitters/maxmin_split.py

```diff
@@ -0,0 +1,291 @@
+
+    References
+    ----------
+    .. [1] `Mark Ashton, John Barnardb, Florence Casset, Michael Charlton, Geoffrey Downsb,
+        Dominique Gorse, JohnHolliday, Roger Lahanac, Peter Willett
+        "Identification of Diverse Database Subsets using Property-Based and Fragment-Based Molecular Descriptions"
+        <https://onlinelibrary.wiley.com/doi/10.1002/qsar.200290002>_`
+    .. [2] `Roger Sayle
+        "Improved RDKit implementation"
+        <https://github.com/rdkit/UGM_2017/blob/master/Presentations/Sayle_RDKitDiversity_Berlin17.pdf>_`
+    .. [3] `Tim Dudgeon
+        "Revisting the MaxMinPicker"
+        <https://rdkit.org/docs/cppapi/classRDPickers_1_1MaxMinPicker.html>_`
+    .. [4] `RDKit MaxMin Picker
+        <https://squonk.it/docs/cells/RDKit%20MaxMin%20Picker>_`
+    """
+    data_size = len(data)
+    train_size, test_size = validate_train_test_split_sizes(
+        train_size, test_size, data_size
+    )
+
+    mols = ensure_mols(data)
+    fps = GetMorganGenerator().GetFingerprints(mols)
+
+    picker = MaxMinPicker()
+    test_idxs = picker.LazyBitVectorPick(
+        fps,
+        poolSize=data_size,
+        pickSize=test_size,
+        seed=random_state,
+    )
+    train_idxs = list(set(range(data_size)) - set(test_idxs))
+
+    train_subset: list[Any] = []
+    test_subset: list[Any] = []
+
+    if return_indices:
+        train_subset = train_idxs
+        test_subset = test_idxs
+    else:
+        train_subset = get_data_from_indices(data, train_idxs)
+        test_subset = get_data_from_indices(data, test_idxs)
+
+    ensure_nonempty_subset(train_subset, "train")
+    ensure_nonempty_subset(test_subset, "test")
+
+    if additional_data:
+        additional_data_split: list[Sequence[Any]] = split_additional_data(
+            list(additional_data), train_idxs, test_idxs
+        )
+        return train_subset, test_subset, additional_data_split
+    else:
+        return train_subset, test_subset
+
+
+@validate_params(
+    {
+        "data": ["array-like"],
+        "additional_data": ["tuple"],
+        "train_size": [
+            Interval(RealNotInt, 0, 1, closed="neither"),
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "test_size": [
+            Interval(RealNotInt, 0, 1, closed="neither"),
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "return_indices": ["boolean"],
+    },
+    prefer_skip_nested_validation=True,
+)
+def maxmin_train_valid_test_split(
+    data: Sequence[Union[str, Mol]],
+    *additional_data: Sequence,
+    train_size: Optional[float] = None,
+    valid_size: Optional[float] = None,
+    test_size: Optional[float] = None,
+    return_indices: bool = False,
+    random_state: int = 0,
+) -> Union[
+    tuple[
+        Sequence[Union[str, Mol]], Sequence[Union[str, Mol]], Sequence[Sequence[Any]]
+    ],
+    tuple[Sequence, ...],
+    tuple[Sequence[int], Sequence[int]],
+]:
+    """
```

Comment:
Implement formatting like for train-test split, both description comments and empty lines (e.g. between references)